File: benchmarks/src/main/ruby/dynamicobjects.rb
require 'java'

def run
  obj = Object.new
  obj.instance_variable_set("@random", java.util.Random.new)
  def obj.rand
    @random.nextInt()
  end
  obj.instance_variable_set("@acc", 0)
  def obj.acc
    @acc
  end
  def obj.acc=(value)
    @acc = value
  end
  i = 0
  while i < 5000000 do
    obj.acc = obj.acc + obj.rand()
    i = i + 1
  end
  return obj.acc
end

run

File: benchmarks/src/main/js/dynamicobjects.js
function run() {
  var obj = new Object();
  obj.acc = 0;
  var random = new java.util.Random();
  obj.rand = function() { return random.nextInt(); }
  for (var i = 0; i < 5000000; i = i + 1) {
    obj.acc = obj.acc + obj.rand();
  }
  return obj.acc;
}